integration-tests/push-to-external-registry-idempotent/test.sh

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env bash
+#
+# test.sh - Test-specific functions for push-to-external-registry-idempotent
+#
+# This file contains functions for the idempotent two-release test flow.
+# It overrides wait_for_releases() to create two releases with the same snapshot.
+#
+
+# Source the base test.sh for shared functions
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "${SCRIPT_DIR}/../push-to-external-registry/test.sh"
+
+# --- Global Script Variables (Defaults) ---
+# CLEANUP inherited from base test.sh
+
+# Override wait_for_releases to create TWO releases with the same snapshot
+# This function validates idempotent behavior
+wait_for_releases() {
+    echo "════════════════════════════════════════════════════════════════════"
+    echo "IDEMPOTENT TEST: Creating Two Releases with Same Snapshot"
+    echo "════════════════════════════════════════════════════════════════════"
+    
+    # Wait for snapshot to appear (up to 12 minutes)
+    echo "Waiting for snapshot to appear..."
+    local SNAPSHOT_NAME=""
+    local max_attempts=72  # 12 minutes with 10-second intervals
+    local attempt=1
+
+    while [ $attempt -le $max_attempts ] && [ -z "$SNAPSHOT_NAME" ]; do
+        SNAPSHOT_NAME=$(kubectl get snapshots -n "${tenant_namespace}" \
+            --sort-by=.metadata.creationTimestamp \
+            -l appstudio.openshift.io/application="${application_name}" \
+            -o jsonpath='{.items[-1].metadata.name}' 2>/dev/null || true)
+        
+        if [ -n "${SNAPSHOT_NAME}" ]; then
+            echo "✅ Found snapshot: ${SNAPSHOT_NAME}"
+            break
+        fi
+        
+        echo "  Attempt $attempt/$max_attempts: No snapshot found yet, waiting 10 seconds..."
+        sleep 10
+        attempt=$((attempt + 1))
+    done
+
+    if [ -z "${SNAPSHOT_NAME}" ]; then
+        echo "❌ ERROR: No snapshot found after waiting $((max_attempts * 10)) seconds"
+        echo "Available snapshots in namespace ${tenant_namespace}:"
+        kubectl get snapshots -n "${tenant_namespace}" \
+            -l appstudio.openshift.io/application="${application_name}" \
+            -o wide 2>/dev/null || echo "  (none found)"
+        exit 1
+    fi
+
+    echo "Using snapshot: ${SNAPSHOT_NAME}"
+    echo ""
+
+    # Create unique release names
+    local RELEASE_1_NAME="release-idempotent-1-${uuid}"
+    local RELEASE_2_NAME="release-idempotent-2-${uuid}"
+
+    # PHASE 1: First Release (Initial)
+    echo "════════════════════════════════════════════════════════════════════"
+    echo "PHASE 1: First Release (Initial)"
+    echo "════════════════════════════════════════════════════════════════════"
+    
+    echo "Creating first release..."
+    local RELEASE_1_START=$(date +%s)
+
+    cat <<EOF | kubectl apply -f -
```

Comment:
There is already a release created automatically. Why create two new releases again?